OIDC and OIDC Client: Support JWT bearer client authentication using client assertion loaded from filesystem (#45131)
quarkusio/quarkus: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc

Next, the JWT bearer token must be provided as a `client_assertion` parameter to the OIDC client.

> Quarkus can load the `client_assertion` from a file system.

Reviewer: ```suggestion
Quarkus can load the JWT bearer token from a file system.
```

> For example in Kubernetes, a service account token projection can be mounted to a `/var/run/secrets/tokens` path.

Reviewer: ```suggestion
For example, in Kubernetes, a service account token projection can be mounted to a `/var/run/secrets/tokens` path.
```

Then all you need to do is configure a JWT bearer token path as follows:

[source,properties]
----
quarkus.oidc-client.credentials.jwt.jwt-bearer-token-path=/var/run/secrets/tokens <1>
----

> <1> Path to a JWT Bearer token. Quarkus loads a new token from a filesystem when necessary according to the token expiration.

Reviewer: ```suggestion
<1> Path to a JWT bearer token. Quarkus loads a new token from a filesystem and reload it when the token has expired.
```
